Automattic/jetpack
# Button Block: ensure no borders get added by default browser rules

Fixes https://github.com/Automattic/themes/issues/6068
Fixes #25018

#### Changes proposed in this Pull Request:

This addresses an issue that was brought up in https://github.com/Automattic/themes/issues/6068

When themes do not include any styles that reset the default `button` browser styles, the Jetpack button block will display borders on the frontend. To avoid that, we add our own reset, except when the button uses the "outline" styles, where we do want to show a border.

**The danger here is that those styles will overwrite button border styles that may have been set by the theme.**

#### Jetpack product discussion

* N/A

#### Does this pull request change what data or activity we track or use?

* No

#### Testing instructions:

* Start with a site using the Twenty Twenty Two theme.
* In a new post, add 2 blocks that use the Jetpack button block: the form block as well as the Revue block are good examples.
* In one of those blocks, use the default "fill" block style, and in the other use the "outline" style.
* On the frontend, the button should not have a border when using the "fill" block style.
